evm/src/main/java/org/hyperledger/besu/evm/frame/MessageFrame.java

```diff
@@ -684,6 +700,22 @@ private void setUpdatedMemory(
     }
   }
 
+  private void setUpdatedMemory32ByteAligned(
+      final long offset, final long length, final Bytes value) {
+    if (length > 0) {
+      final int srcSize = value.size();
+      if (length > srcSize) {
+        final MutableBytes paddedAnswer = MutableBytes.create((int) length);
+        if ((long) 0 < srcSize) {
+          value.slice(0, (int) ((long) srcSize)).copyTo(paddedAnswer, (int) (length - srcSize));
```

Comment:
what is casting srcSize to long and then back to int doing when defining the length of the slice being copied?

Reply:
Not sure how neither IntelliJ nor Errorprone tagged it.  Initially it was because memory was going to be long indexed, until I was satisfied that 33rd bit memory access was unlikely to ever be viable even in perverse circumstances.

Reply:
Removed.  Sonar flagged it.